backend/internal/storage/postgres/storage.go

```diff
@@ -0,0 +1,51 @@
+package storage
+
+import (
+	"backend/internal/config"
+	"context"
+	"log"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+// Repository storage of all repositories.
+type Repository struct {
+	DB *pgxpool.Pool
+}
+
+// Closes the pooled connection for cleanup and resource management.
+func (r *Repository) Close() error {
+	r.DB.Close()
+	return nil
+}
+
+// Establishes a sustained connection to the PostgreSQL database using pooling.
+func ConnectDatabase(config config.DB) *pgxpool.Pool {
```

Comment:
this provides more flexibility to the caller

```suggestion
func ConnectDatabase(ctx context.Context, config config.DB) (*pgxpool.Pool, error) {
```